### PR TITLE
[GAIA Compiler] Build algebra layer structures for group and order in `GraphBuilder`

### PR DESCRIPTION
**Pull request by shirly121**
## What do these changes do?
1. build algebra layer structures for group and order in `GraphBuilder`.
2. fix bugs in `GraphSchemaTypeList`, which contains schema meta for a vertex or an edge in fuzzy conditions.

## Related issue number

Fixes

### CODE REVIEW

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rex/RexTmpVariable.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexVisitor;
+import org.apache.commons.lang3.StringUtils;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.Objects;
+
+/**
+ * maintain the necessary info to create a {@code RexGraphVariable}
```
Comment (longbinlai) [inline interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rex/RexTmpVariable.java:30]: Please describe why this class is necessary

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rex/RexVariableConverter.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+package com.alibaba.graphscope.common.ir.rex;
+
+import com.alibaba.graphscope.common.ir.tools.GraphBuilder;
+
+import org.apache.calcite.rex.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+// build from RexTmpVariable to RexGraphVariable
```
Comment (longbinlai) [inline interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rex/RexVariableConverter.java:26]: {@code RexTmpVariable} and {@code RexGraphVariable}

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/GroupTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                                                GraphStdOperatorTable.PLUS,
+                                                builder.variable(null, "age"),
+                                                builder.literal(1))))
+                        .build();
+        Assert.assertEquals(
+                "GraphLogicalAggregate(keys=[{variables=[DEFAULT.name], aliases=[name]}],"
+                    + " values=[[{operands=[$f0], aggFunction=COUNT, alias='x', distinct=true}]])\n"
+                    + "  GraphLogicalProject($f0=[+(DEFAULT.age, 1)], isAppend=[true])\n"
+                    + "    GraphLogicalSource(tableConfig=[{isAll=false, tables=[person]}],"
+                    + " alias=[~DEFAULT], opt=[VERTEX])",
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/GroupTest.java:81]: Unify the ~DEFAULT print

File: interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/GroupTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                        + " b]}], values=[[{operands=[DEFAULT], aggFunction=COUNT, alias='c',"
+                        + " distinct=false}]])\n"
+                        + "  GraphLogicalSource(tableConfig=[{isAll=false, tables=[person]}],"
+                        + " alias=[~DEFAULT], opt=[VERTEX])",
+                aggregate.explain().trim());
+    }
+
+    // group by HEAD.name, count(HEAD.age+1, 'x') -> project({HEAD.age+1 as '$f0'}, isAppend = true)
+    // + aggregate(keys={HEAD.name}, calls=[count($f0) as 'x'])
+    @Test
```
Comment (longbinlai) [inline interactive_engine/compiler/src/test/java/com/alibaba/graphscope/common/ir/GroupTest.java:58]: Refine the comments and add another test case for when agg key is an expression.

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/type/group/AbstractAggCall.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+package com.alibaba.graphscope.common.ir.rel.type.group;
+
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.tools.RelBuilder;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * implement interfaces of {@code AggCall} by default, these interfaces are useless to {@code GraphAggCall}
+ */
+public abstract class AbstractAggCall implements RelBuilder.AggCall {
```
[the remark this answers is not in the document]
Comment (shirly121) [reply interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/type/group/AbstractAggCall.java]: > Have no idea what AbstractAggCall is different from AggCall.

all fixed in commit 5bdf04403f99c991cc6ba959190e352a3eacf27e

File: interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/type/group/GraphGroupKeys.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * maintains group keys and aliases
+ */
+public class GraphGroupKeys implements RelBuilder.GroupKey {
+    private final List<RexNode> variables;
+    private final List<@Nullable String> aliases;
+
+    public GraphGroupKeys(List<RexNode> variables, List<@Nullable String> aliases) {
+        this.variables = Objects.requireNonNull(variables);
```
[the remark this answers is not in the document]
Comment (shirly121) [reply interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/rel/type/group/GraphGroupKeys.java:38]: > 这两个地方需要检查长度一样吗？

不强制用户给一样长度，对于aliases.size() < variables.size()情况，会内部自动为最后的variables生成aliases

**Review by longbinlai — APPROVED** [review]
LGTM

**Pull request merged**